### PR TITLE
DR-110: CI/CD + project configuration

### PR DESCRIPTION
**Pull request by thathert**
This is still WIP. I will add my notes around getting projects set up and fix Migrate.java so that it decides whether to `dropAll` based on a configuration property. 

This configures Travis to do the following steps (It will fail the build if any step fails):

1. assemble: generate code, etc
2. check: lint, findbugs, unit tests* (we can probably run connected tests now)
3. build docker container and push to GCR
4. deploy that image to GKE in the broad-jade-integration

This required me to get the Google Cloud SDK working on Travis. There is an older version there that we shouldn't use. I provided service account credentials for travis-access@broad-jade-dev... as a Travis secure variable. The job will write this to a json file and then authenticate with the SDK using it.

The travis-access SA has a role that allows it to create objects in cloud storage (needed for pushing to gcr) and a role that allows it to deploy to both the integration and dev GKEs. Both GKE's are running under a project specific jade-k8-sa service account that has access to the resources in the project.

I'll include the documentation and setup scripts for these in the `ops/` folder.

Note: This doesn't include any actual integration tests yet. I'd like to do that as a separate PR.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -239,7 +239,7 @@ def getGitHash = { ->
 
 // TODO: don't have access to broadinstitute to create a repository, or is there a better place?
 docker {
-    name "jhert/${bootJar.baseName}:${getGitHash()}"
+    name "gcr.io/broad-jade-dev/${bootJar.baseName}:${getGitHash()}"
```
Comment (ddietterich) [inline build.gradle:241]: Nit: is the TODO still right?

File: ops/wait-for-deployment.sh [hunk 2]
```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+# Waits for a deployment to complete.
+#
+# Includes a two-step approach:
+#
+# 1. Wait for the observed generation to match the specified one.
+# 2. Waits for the number of available replicas to match the specified one.
+#
+# Spawn from my answer to this StackOverflow question: http://stackoverflow.com/questions/37448357/ensure-kubernetes-deployment-has-completed-and-all-pods-are-updated-and-availabl
+#
```
Comment (ddietterich) [inline ops/wait-for-deployment.sh:10]: There is a bash idiom to make a log of your script into a file. Something like:
```
exec > logfile  # replace the current shell with a new shell writing stdout to logfile
exec 2>&1       # send stderr there also
set -x          # echo script commands to stdout, so you see the whole flow in the script
```
might be useful for debugging?

**Review by ddietterich — APPROVED** [review]
LGTM

**Review by RoriCremer — APPROVED** [review]

**Review by ahaessly — APPROVED** [review]
👍

**Pull request merged**